src/libsyntax/parse/attr.rs

```diff
@@ -35,7 +35,10 @@ impl<'a> Parser<'a> {
                     self.span.hi
                 );
                     if attr.node.style != ast::AttrStyle::Outer {
-                        return Err(self.fatal("expected outer comment"));
+                        let mut err = self.fatal("expected outer comment");
+                        err.fileline_note(self.span, "inner comments like this must \
+                                                      be placed before any items");
```

Comment:
good point. I think it'll have to be the example to remain terse enough. Should probably also say "doc comment" instead of just "comment"...